apps/backoffice-v2/src/pages/Entity/hooks/useTasks/hooks/useDirectorsBlocks/useDirectorsBlocks.tsx

```diff
@@ -0,0 +1,242 @@
+import { AnyObject } from '@ballerine/ui';
+import { UseQueryResult } from '@tanstack/react-query';
+import { useCallback, useMemo } from 'react';
+import { toTitleCase } from 'string-ts';
+import { valueOrNA } from '../../../../../../common/utils/value-or-na/value-or-na';
+import { TWorkflowById } from '../../../../../../domains/workflows/fetchers';
+import { composePickableCategoryType } from '../../../useEntity/utils';
+import { useRemoveDecisionTaskByIdMutation } from '../../../../../../domains/entities/hooks/mutations/useRemoveDecisionTaskByIdMutation/useRemoveDecisionTaskByIdMutation';
+import { getPostUpdateEventName } from '../../get-post-update-event-name';
+import { selectDirectorsDocuments } from '../../selectors/selectDirectorsDocuments';
+import { useAuthenticatedUserQuery } from '../../../../../../domains/auth/hooks/queries/useAuthenticatedUserQuery/useAuthenticatedUserQuery';
+import { useCaseState } from '../../../../../../pages/Entity/components/Case/hooks/useCaseState/useCaseState';
+import { StateTag } from '@ballerine/common';
+import { motionProps } from '../../motion-props';
+
+export type Director = AnyObject;
+
+export const useDirectorsBlocks = (
+  workflow: TWorkflowById,
+  documentFiles: UseQueryResult[],
+  documentImages: Array<Array<string>>,
+) => {
+  const { mutate } = useRemoveDecisionTaskByIdMutation(
+    workflow.id,
+    getPostUpdateEventName(workflow),
+  );
+  const { data: session } = useAuthenticatedUserQuery();
+  const caseState = useCaseState(session?.user, workflow);
+
+  const directors: Director[] = useMemo(
```

Comment:
Type is already specified once.
```suggestion
  const directors = useMemo(
```